packages/base/src/panelview/objecttree.tsx

```diff
@@ -464,8 +468,61 @@ class ObjectTreeReact extends React.Component<IProps, IStates> {
                       textOverflow: 'ellipsis',
                       overflowX: 'hidden'
                     }}
+                    onClick={() => this.setState({ editingNode: opts.node.id })}
                   >
-                    {opts.node.label}
+                    {this.state.editingNode === opts.node.id ? (
+                      <input
+                        type="text"
+                        value={this.state.editingNodeValue || opts.node.label}
+                        autoFocus
+                        onBlur={() => this.setState({ editingNode: null })}
+                        onChange={e =>
+                          this.setState({ editingNodeValue: e.target.value })
+                        }
+                        onKeyDown={e => {
+                          if (e.key === 'Enter') {
+                            console.log(
+                              'Updated name:',
+                              this.state.editingNodeValue
+                            );
+                            const updatedName = this.state.editingNodeValue;
+                            const objectName = opts.node.id;
+
+                            if (this.props.cpModel.mainViewModel) {
+                              const obj =
+                                this.props.cpModel.jcadModel?.sharedModel.getObjectByName(
+                                  objectName as string
+                                );
+                              if (obj) {
+                                this.props.cpModel.sharedModel?.updateObjectByName(
```

Comment:
We not only need to update the specific object, but we also need to go through all objects and see if they depend on it. All objects have a `dependency` parameter which is a list of object it depends on, by name.

I believe it's easier to tackle this renaming task once https://github.com/jupytercad/JupyterCAD/issues/200 is done. We wouldn't need to do that work I'm describing above because the list of dependencies would be a list of UUIDs.

I would suggest we put this renaming PR on hold while https://github.com/jupytercad/JupyterCAD/issues/200 is not tackled.